### PR TITLE
Make contributing guide more friendly

### PR DESCRIPTION
**Pull request by akshayjshah**
Update `CONTRIBUTING.md` to (1) reflect changes in Fx's build tooling,
and (2) hand-hold a bit more.

The current guide is pretty focused on nuts and bolts that most people
won't run into, and it glosses over some important details: how to clone
the repo, how to create a feature branch, etc. Basically, experienced
Gophers and git users don't need any of this, and newbies need more
hand-holding.

Along the way, I've tried to strike a more welcoming tone. Rather than a
laundry list of the things contributors must do to avoid having a PR
rejected, I've tried to focus on how they can be successful.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,84 +1,80 @@
...
-// FIXME(ai) make this actually work
+```
+make test
+make lint
 ```
 
-Your initials in parens are optional but good practice. This is better
-than a TODO because our CI checks for unresolved FIXMEs. If you forget to fix
-a FIXME, your code won't get merged.
+If you're not using the minor version of Go specified in the Makefile's
```
Comment (Lupie) [inline CONTRIBUTING.md]: I'd phrase this in the positive, so like:
For `make lint` to work, you must be using the minor version of Go specified in the Makefile's `LINTABLE_MINOR_VERSIONS`. If you're using another version, that's fine, but you'll only discover style issues after opening your pull request.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,84 +1,80 @@
...
+`LINTABLE_MINOR_VERSIONS` variable, `make lint` doesn't do anything. This is
+fine, but it means that you'll only discover style violations after you open
+your pull request.
 
-## Testing
+## Making Changes
```
Comment (Lupie) [inline CONTRIBUTING.md]: Let me double-check, but I think we've agreed on sentence case for H2s and smaller. Title case only for H1s.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,84 +1,80 @@
...
+git checkout master
+git fetch upstream
+git rebase upstream/master
+git checkout -b cool_new_feature
 ```
 
-### Disabling the race detector
-
-The race detector makes the tests run way slower. To disable it:
+Make your changes, then ensure that `make lint` and `make test` still pass. If
```
Comment (Lupie) [inline CONTRIBUTING.md]: Make your changes, and then check that

File: CONTRIBUTING.md (hunk 2 above)
Comment (Lupie) [inline CONTRIBUTING.md]: Ya, GitHub lacks a real set standard on style like this, but it definitely leans toward H1s = title, H2s and below = sentence. I'm probably the only one who cares, but I'm for embracing this rule consistently.

File: CONTRIBUTING.md (hunk 1 above)
Comment (akshayjshah) [inline CONTRIBUTING.md]: Sounds good.

File: CONTRIBUTING.md (hunk 2 above)
Comment (akshayjshah) [inline CONTRIBUTING.md]: Fair enough, happy to comply.

File: CONTRIBUTING.md (hunk 3 above)
Comment (akshayjshah) [inline CONTRIBUTING.md]: :+1:

**Pull request merged**